_posts/2018-05-18-contributing-to-installer-puppet-modules.md

```diff
@@ -0,0 +1,326 @@
+[ WARN 2018-05-16T16:50:10 verbose] +++ /tmp/puppet-file20180516-622-ej00eh     2018-05-16 16:50:10.657060958 +0000
+[ WARN 2018-05-16T16:50:10 verbose] @@ -9,5 +9,5 @@
+[ WARN 2018-05-16T16:50:10 verbose]    <IfModule mod_passenger.c>
+[ WARN 2018-05-16T16:50:10 verbose]      PassengerEnabled off
+[ WARN 2018-05-16T16:50:10 verbose]    </IfModule>
+[ WARN 2018-05-16T16:50:10 verbose] -  Options +FollowSymlinks
+[ WARN 2018-05-16T16:50:10 verbose] +  Options +FollowSymlinks +Indexes
+[ WARN 2018-05-16T16:50:10 verbose]  </Location>
+```
+
+Looks good! Our total code change was:
+
+```diff
+[vagrant@centos7-katello-nightly puppet-katello]$ git diff --cached
+diff --git a/files/pulp-apache-ssl.conf b/files/pulp-apache-ssl.conf
+deleted file mode 100644
+index f08ddfe..0000000
+--- a/files/pulp-apache-ssl.conf
++++ /dev/null
+@@ -1,13 +0,0 @@
+-### File managed with puppet ###
+-
+-<Location /pulp/api>
+-  SSLUsername SSL_CLIENT_S_DN_CN
+-</Location>
+-
+-Alias /pub /var/www/html/pub
+-<Location /pub>
+-  <IfModule mod_passenger.c>
+-    PassengerEnabled off
+-  </IfModule>
+-  Options +FollowSymLinks +Indexes
+-</Location>
+diff --git a/manifests/pulp.pp b/manifests/pulp.pp
+index d0d7b95..ac1174e 100644
+--- a/manifests/pulp.pp
++++ b/manifests/pulp.pp
+@@ -32,6 +32,7 @@ class katello::pulp (
+   Boolean $db_unsafe_autoretry = $::katello::pulp_db_unsafe_autoretry,
+   Optional[Enum['majority', 'all']] $db_write_concern = $::katello::pulp_db_write_concern,
+   Boolean $manage_db = $::katello::pulp_manage_db,
++  String $pub_dir_options = "+FollowSymlinks +Indexes",
+ ) {
+   include ::certs
+   include ::certs::qpid_client
+@@ -86,7 +87,7 @@ class katello::pulp (
+ 
+   foreman::config::passenger::fragment { 'pulp':
+     content     => file('katello/pulp-apache.conf'),
+-    ssl_content => file('katello/pulp-apache-ssl.conf'),
++    ssl_content => template('katello/pulp-apache-ssl.conf.erb'),
+   }
+ 
+   # NB: we define this here to avoid a dependency cycle. It is not a problem if
+diff --git a/templates/pulp-apache-ssl.conf.erb b/templates/pulp-apache-ssl.conf.erb
+new file mode 100644
+index 0000000..2820a8b
+--- /dev/null
++++ b/templates/pulp-apache-ssl.conf.erb
+@@ -0,0 +1,13 @@
++### File managed with puppet ###
++
++<Location /pulp/api>
++  SSLUsername SSL_CLIENT_S_DN_CN
++</Location>
++
++Alias /pub /var/www/html/pub
++<Location /pub>
++  <IfModule mod_passenger.c>
++    PassengerEnabled off
++  </IfModule>
++  Options <%= @pub_dir_options %>
++</Location>
+```
+
+Of course, this only covers port 443 and not 80. You'll need to do something similar to make port 80 work.
+
+After all that is done, you can commit your change and create a pull request.  Each module has its own process for this. You'll find more information in `CONTRIBUTING.md`, which will outline how to run unit tests. Once those are successful, you can put in a pull request.
```

Comment:
There are some changes to the installer that are complex enough that you need to test it from a fresh run of the whole thing. Thus, I think it could be instructive to add something like putting it all together and testing your PR fresh using https://github.com/theforeman/forklift/blob/master/docs/development.md#test-puppet-module-pull-requests